.gitignore

```diff
@@ -3,3 +3,5 @@ layers/
 data/
 *.xml
 node_modules/
+.vagrant/
+Vagrantfile
```

Comment:
Yeah, you'll have to commit or stash your changes, otherwise git will get vocal. Stashing would be the most convenient, then you can just pop wherever you checkout, and potentially `git update-index --assume-unchanged Vagrantfile` if you'd like to tell git to be quiet about it.

An alternative approach is to never commit the Vagrantfile, and instead have a Vagrantfile.default or similarly that a user would have to rename to Vagrantfile. That sort of makes a simple thing somewhat more complex, but also can avoid minor things down the road.